init/eessi_environment_variables

```diff
@@ -81,6 +81,8 @@ if [ -d $EESSI_PREFIX ]; then
         if [ ! -z $EESSI_BASIC_ENV ]; then
           show_msg "Only setting up basic environment, so we're done"
         elif [ -d $EESSI_SOFTWARE_PATH ]; then
+          EESSI_SITE_INSTALLPATH=${EESSI_SOFTWARE_PATH/versions/host_injections}
+          show_msg "Using ${EESSI_SITE_INSTALLPATH} as the site extension directory for installations."
```

Comment:
Lol, I said it myself to you on chat, and then I didn't do it.

Reply:
Actually: the EESSI_MODULE_SUBDIR isn't set if a `EESSI_CUSTOM_MODULEPATH` is set.

Reply:
It's actually a question if `export EESSI_SITE_MODULEPATH=${EESSI_MODULEPATH/versions/host_injections}` still works if a `EESSI_CUSTOM_MODULEPATH` is set. Because in that case, `EESSI_MODULPATH=$EESSI_CUSTOM_MODULEPATH`. I'm not sure where that custom MNS is supposed to be: in the EESSI repo? I.e. is it just a way for users to switch between MNS-es that we support? Or is it going to be a path to a local filesystem somewhere? (in which case the replacement of `versions` with `host_injections` does not give you a sensible `EESSI_SITE_MODULEPATH`. There might not even be a `versions` keyword in there`.

Reply:
Hm, easiest solution might be to also set `EESSI_MODULE_SUBDIR` outside of the if-else statement for `if [ -z ${EESSI_CUSTOM_MODULEPATH+x} ]; then`. That way, this variable is always defined (and used to set the `EESSI_SITE_MODULEPATH`. But, if ` EESSI_CUSTOM_MODULEPATH` is used, _that_ is used as `EESSI_MODULEPATH`.

Reply:
Ok, I went for that approach, see commit https://github.com/EESSI/software-layer/pull/778/commits/d209b272837b60bed6dd221e74eb323fa7b8009f